src/core/issues-notes.js

```diff
@@ -141,31 +141,32 @@ export function run(conf, doc, cb) {
     issueBase = conf.issueBase;
   if ($ins.length) {
     if (conf.githubAPI) {
-      ghFetch(conf.githubAPI)
-        .then(function(json) {
+      (async function(githubAPI) {
+        try {
+          let json = await ghFetch(githubAPI);
           issueBase = issueBase || json.html_url + "/issues/";
-          return fetchIndex(json.issues_url, {
+          let issues = fetchIndex(json.issues_url, {
             // Get back HTML content instead of markdown
             // See: https://developer.github.com/v3/media/
-            Accept: "application/vnd.github.v3.html+json",    
+            Accept: "application/vnd.github.v3.html+json",
           });
-        })
-        .then(function(issues) {
+          
```

Comment:
I see that you've broken the whole .then chain and converted it to behave like a synchronous function. LGTM. I just feel that we could modularize the code better so that maintaining it would become easier.

Reply:
For what it's worth, @OrionStar25, you are correct... this whole module is terribly written. Part of this simple exercise was to see how people deal with jumping into fairly bad code to make it better.